### ISSUE TITLE
Ads are shown nevertheless...

### ISSUE THREAD

**Issue by Gnuop** [gorhill/uBlock#1044]
on focus.de, prosieben.de, stern.de, wetter.com, finanzen.net, handy-inside.de and many more.

http://t3n.de/news/adblocker-adblock-defender-600335/ (its german, try google translate)

more clever system than easylist needs to be established. on the long run every website will have these mechanisms, big companies are supporting it in germany

**Comment by gorhill** [comment]
Please open specific issues and details on how to reproduce. I have no idea what to do with what is reported here, as there is nothing specifc, no details, no filter lists, no repro steps, etc.

**Comment by gorhill** [comment]
![d](https://cloud.githubusercontent.com/assets/585534/6708493/497a9dd0-cd47-11e4-8e72-e75ababdc300.png)

**Comment by my-password-is-password** [comment]
dup of #380?

**Comment by gorhill** [comment]
I could remove the ads in the screenshot above, using the filter in the element picker. Ot worked. It's just a matter of creating a new filter when the site workaround the older one. It's a filter lists maintenance thing.

**Comment by tlu1024** [comment]
Agreed. It's usually just necessary to place an asterisk at the right place in the filter offered by the Element Picker. Examples concerning some affected sites:

||bilder.pcwelt.de/*_03100x07680.jpg

||focus.de/*crop*.gif

||www.prosieben.de/static/*.gif

||www.inside-handy.de/image/*.gif

**Comment by Gnuop** [comment]
the link above translated: https://translate.google.com/translate?sl=auto&tl=en&js=y&prev=_t&hl=en&ie=UTF-8&u=http%3A%2F%2Ft3n.de%2Fnews%2Fadblocker-adblock-defender-600335%2F&edit-text=

its worth reading it...you'll recognize theres a clever technology behind it.

of course its just a matter of creating a new filter. but thats not the way it works for the general, non technical user. He doesnt know how to do that. 
If the filter is listed in any lists the anti-adblocker-software tests against, it is not working any longer cause the anti-adblocker software changes the path, the format (jpg oder gif) and creates a new name for the image file.
its a clever system. more and more sites are supporting it: many really BIG sites are using it like finanzen.net, stern.de or focus.de. the last 2 sites or the biggest magazine sites in germany. prosieben.de is one of the biggest TV stations.

the product listed in the article belongs to one of the greatest publisher in germany. International companies are already making there own products.

i think the filter system isnt working and not a good solution against these challenges - you can see that in the easylist changes. sometimes 2 hours after changing the filter it wasnt working any longer.

what do you think could be done?

perhaps not showing any pictures with scrambled names?
perhaps not showing pictures on those positions, where these ads are shown?
perhaps you have better solutions?

**Comment by tlu1024** [comment]
> of course its just a matter of creating a new filter. but thats not the way it works for the general, non technical user. He doesnt know how to do that.

Well, many of the already existing filters are relatively cryptic for someone who isn't familiar with the ABP filter syntax. That's why gorhill said that it's a filter lists maintenance thing. 

> i think the filter system isnt working and not a good solution against these challenges

This remains to be seen. So far, I doubt that you're right. I created the filters in my previous post yesterday and they still work today. I suggest that you report unblocked content and ask for filters in the respective sub-forums on https://forums.lanik.us/

**Comment by gorhill** [comment]
> thats not the way it works for the general, non technical user. He doesnt know how to do that. 

Last commit is to try to alleviate this.

- Launch element picker.
- Click one of the ad.
- Element picker dialog opens, with one choice of network filter: the image just clicked.
- Go back to pick again -- by clicking "Pick" or by clicking outside the dialog box.
- Click on **another** ad image.
- Element picker dialog opens, withnow two choices of network filters:
    - The image just clicked; and
    - A new network filter which work for the image just clicked **and** the image previously clicked.

Most often, this "composite" filter will end up selecting all the ads on the page. But it is incremental, meaning if a user keep clicking on images, the suggested composite filter will become more and more broad.

So this alleviate the need for a user to understand filter syntax. And even for more experienced users, this is nice, the element picker does automatically what the user would end up doing anyways after searching for patterns in multiple URLs.

**Comment by harshanvn** [comment]
Wow, this is a nice addition to counter these kind of ads.

**Comment by Gnuop** [comment]
@gorhill : thanks for your description. i was hoping that there would be a solution which does not involve the end user but instead gets down to the root of the trouble. 
the casual/even general user won't do that and in the end he could even think the adblocker doesnt work. but perhaps i am wrong! 
I expext the publishers to continue to integrate the anti adblocker software cause they earn a lot of money (which shows that no one/a minority is making individual filters). 

btw: your decribed method doesnt work on www.tvspielfilm.de

@tlu1024:
"This remains to be seen. So far, I doubt that you're right. I created the filters in my previous post yesterday and they still work today. I suggest that you report unblocked content and ask for filters in the respective sub-forums on https://forums.lanik.us/"
-> the maintainer of the geman easylist gave up. check out the geman easylist forum. 

if ublock would find a solution in its blockers itself it would be a really great unique feature compared to adblock plus.

**Comment by gorhill** [comment]
> your decribed method doesnt work on www.tvspielfilm.de

Yeah I was just thinking earlier that I should make the method work across page load -- currently it work only for one session element picker session. This is not difficult to implement, as always it's more about finding the proper UI. I will work toward the solution working across page reload, i.e. launch element picker, select the ads, force a page reload, launch the element picker, pick more ads, by then it should be sufficient for uBlock to figure a proper filter. Well it would be for the case above.

But it's just one more tool, not a guarantee to solve the never ending problem: it's a cat and mouse game, no javascript can replace a human in identifying what is an ad and what is not. 

The ultimate tools is the user: when a site ends up peppering itself with unavoidable ads, users should just stop going to that site, that will solve the problem. users have the final word.

By the way, these particular ads are really ruining their sites, it's amazing to see they think they are a great idea.

Anyway, I will just further fine tune that tool.

**Comment by tlu1024** [comment]
> the maintainer of the geman easylist gave up. check out the geman easylist forum.

I did. Where did he say this? On the contrary I'm seeing a lot of proposed filters for thes sites mentioned by you and others.

@gorhill : This is an excellent enhancement - congratulations!

**Comment by gorhill** [comment]
> work across page load

Implemented with 45882970fad63736338923079ca696d0ecaa7cb2: <https://www.youtube.com/watch?v=WoFC3EJWg08>.

**Comment by alejandrolemus** [comment]
> @gorhill : This is an excellent enhancement - congratulations!

+1, excellent indeed
